exercises/run-length-encoding/src/example/java/RunLengthEncoding.java

```diff
@@ -4,41 +4,43 @@
 public class RunLengthEncoding {
     public String encode(String data) {
 
-        StringBuilder encodedData = new StringBuilder();
+        StringBuilder encodedDataBuilder = new StringBuilder();
 
         if (data.length() == 0) {
             return data;
         }
 
-        int charCount = 0;
-        char beforeChar = data.charAt(0);
+        int previousCharCount = 1;
+        char previousChar = data.charAt(0);
 
-        for (int i = 0; i < data.length(); i++) {
+        for (int i = 1; i < data.length(); i++) {
             char currentChar = data.charAt(i);
 
-            if (beforeChar == currentChar) {
-                charCount++;
-
-                if (i != data.length() - 1) {
-                    continue;
-                }
+            if (previousChar == currentChar) {
+                previousCharCount++;
             }
 
-            if (charCount != 1) {
-                encodedData.append(charCount);
-            }
+            if (previousChar != currentChar || i == data.length() - 1) {
+                addChars(encodedDataBuilder, previousChar, previousCharCount);
 
-            encodedData.append(beforeChar);
+                if (previousChar != currentChar && i == data.length() - 1) {
+                    addChars(encodedDataBuilder, currentChar, 1);
+                }
 
-            if (beforeChar != currentChar && i == data.length() - 1) {
-                encodedData.append(currentChar);
+                previousChar = currentChar;
+                previousCharCount = 1;
             }
+        }
+
+        return encodedDataBuilder.toString();
+    }
 
-            beforeChar = currentChar;
-            charCount = 1;
+    public void addChars(StringBuilder toAddTo, char toAdd, int countOfChar) {
+        if(countOfChar != 1) {
+            toAddTo.append(countOfChar);
         }
 
-        return encodedData.toString();
+        countOfChar.append(toAdd);

```

Comment:
This is why the build is failing. It should be `toAddTo.append(toAdd)`. Sorry I didn't spot it before!